Assets/Src/Scripts/Input/WebCamSource.cs

```diff
@@ -0,0 +1,203 @@
+    [SerializeField] private RenderTexture _buffer;
+    private WebCamTexture _webCam;
+    private int _width = 1280;
+    private int _height = 720;
+    private static readonly Dictionary<ResolutionOptions, string> _resolutionDic =
+    new Dictionary<ResolutionOptions, string>
+    {
+      { ResolutionOptions.LOW , "1280x720" } , {  ResolutionOptions.MIDDLE, "1600x900" } , { ResolutionOptions.HIGH , "1920x1080" }
+    };
+    private static readonly Dictionary<FpsOptions, string> _fpsDic =
+    new Dictionary<FpsOptions, string>
+    {
+      { FpsOptions.LOW , "30Hz" } , { FpsOptions.HiGH , "60Hz" }
+    };
+    private static readonly Dictionary<FlipOptions, string> _flipDic =
+    new Dictionary<FlipOptions, string>
+    {
+      { FlipOptions.NORMAL , "Normal" } , { FlipOptions.MIRRORING , "Mirroring" }
+    };
+    private List<string> _devicesName = new List<string>();
+    private List<string> _resolutionOpt = new List<string>();
+    private List<string> _fpsOpt = new List<string>();
+    private List<string> _flipOpt = new List<string>();
+
+    // Start button event.
+    public void Play() {
+      if (_webCam == null) {
+        _webCam = new WebCamTexture(DeviceName, _width, _height, Fps);
+        _webCam.Play();
+      }
+    }
+
+    // Stop button event.
+    public void Stop() {
+      if (_webCam != null) {
+        _webCam.Stop();
+        _webCam = null;
+      }
+    }
+
+    // DropdownChange event.
+    public void CameraDropdownChange(int value) {
+      //Debug.Log(_devicesName.ToArray()[value]);
+      if (_devicesName.ToArray()[value] != null) {
+        DeviceName = _devicesName.ToArray()[value];
+      }
+    }
+
+    // DropdownChange event.
+    public void ResolutionDropdownChange(int value) {
+      switch (value) {
+        case (int)ResolutionOptions.LOW:
+          _width = 1280;
+          _height = 720;
+          break;
+        case (int)ResolutionOptions.MIDDLE:
+          _width = 1600;
+          _height = 900;
+          break;
+        case (int)ResolutionOptions.HIGH:
+          _width = 1920;
+          _height = 1080;
+          break;
+        default:
+          break;
+      }
+    }
+
+    // DropdownChange event.
+    public void FpsDropdownChange(int value) {
+      switch (value) {
+        case (int)FpsOptions.LOW:
+          Fps = 60;
+          break;
+        case (int)FpsOptions.HiGH:
+          Fps = 30;
+          break;
+        default:
+          break;
+      }
+    }
+
+    // DropdownChange event.
+    public void FlipDropdownChange(int value) {
+      switch (value) {
+        case (int)FlipOptions.NORMAL:
+          Flip = false;
+          break;
+        case (int)FlipOptions.MIRRORING:
+          Flip = true;
+          break;
+        default:
+          break;
+      }
+    }
+
+    private void InitDropdown(Dropdown target, List<string> optarr, DropdownKey dropdownkey) {
+      target.ClearOptions();
+      switch (dropdownkey) {
+        case DropdownKey.CAMERA:
```

Comment:
You are still repeating the same logic in your extracted method `InitDropdown`, aren't you? Please take a look, the code pieces 153-156, 159-162, 165-168 are doing exactly the same thing, right? Even consider the code piece 149-150, it is doing a subset of the above ones.

How about the following sample code (where you don't need the enum DropdownKey at all. And you need to change your Enum/Dictionary definitions a little bit):

```
public enum XOptions {
  X0 = 0,
  X1 = 1,
};

public enum YOptions {
  Y0 = 0,
  Y1 = 1,
  Y2 = 2,
};

private static readonly SortedDictionary<int, string> _xOptionDict =
    SortedDictionary<int, string> {
      { (int)XOptions.X0, "Sample" },
      { (int)XOptions.X1, "Sample" },
    };

private static readonly SortedDictionary<int, string> _yOptionDict =
    SortedDictionary<int, string> {
      { (int)YOptions.Y0, "Sample" },
      { (int)YOptions.Y1, "Sample" },
      { (int)YOptions.Y2, "Sample" },
    };

private void InitDropdown(Dropdown target,
                          SortedDictionary<int, string> optDict,
                          Action<int> onChangedCallback) {
  target.ClearOptions();
  List<string> optStrings =
      optDict is null ? new List<string>() : new List<string>(optDict.Values);
  target.AddOptions(optStrings);
  target.onValueChanged.AddListener(onChangedCallback);
  target.value = 0;
}

InitDropdown(CameraDropdown, null, CameraDropdownChange);
InitDropdown(ResolutionDropdown, _xOptionDict, ResolutionDropdownChange);
... ...
```

Does this code look clearer?